### PR TITLE
Build: Only fail when versions mismatch in CI environment

### PR DESCRIPTION
**Pull request by petemill**
Fix https://github.com/brave/brave-browser/issues/5208

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by petemill** [comment]
Note that I proved that the `isCI` variable is correctly set by first pushing a commit which output the value of isCI. The result was:
```
22:40:04  > node ./scripts/commands.js build "Release" "--channel=nightly" "--official_build=true" "--target_os=ios"
22:40:04  
22:40:04  /Users/jenkins/jenkins/workspace/brave-browser-build-pr_PR-5209/lib/build.js:9
22:40:04    throw new Error('Is CI Environment :-)')
22:40:04    ^
22:40:04  
22:40:04  Error: Is CI Environment :-)
```
https://staging.ci.brave.com/job/brave-browser-build-pr/job/PR-5209/1/console
Ref: 0115b0f8

File: lib/build.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ const util = require('../lib/util')
 const path = require('path')
 const fs = require('fs-extra')
 
+const isCIEnvironment = !!process.env.CI
```
Comment (mihaiplesa) [inline lib/build.js]: suggesting `buildEnvironment = process.env.BUILD_ENV`

File: lib/build.js [hunk 2]
```diff
@@ -81,7 +83,7 @@ const checkVersionsMatch = () => {
   const braveVersionFromChromeFile = `${found[2]}.${found[3]}.${found[4]}`
   if (braveVersionFromChromeFile !== config.braveVersion) {
     console.warn(`Version files do not match!\nsrc/chrome/VERSION: ${braveVersionFromChromeFile}\nbrave-browser package.json version: ${config.braveVersion}`)
-    if (config.buildConfig === 'Release') {
+    if (isCIEnvironment) {
```
Comment (mihaiplesa) [inline lib/build.js]: to check if `buildEnvironment` is "ci"`

File: Jenkinsfile [hunk 3]
```diff
@@ -16,6 +16,7 @@ pipeline {
         booleanParam(name: "DEBUG", defaultValue: false, description: "")
     }
     environment {
+        CI = 1
```
Comment (mihaiplesa) [inline Jenkinsfile]: can be omitted

**Comment by petemill** [comment]
Considering just removing the check from the repo script entirely since it looks like Jenkinsfile has a very similar check

**Comment by mihaiplesa** [comment]
@petemill sounds good to me, do we close this PR then?

**Comment by petemill** [comment]
@mihaiplesa The aim is the same so I will amend the branch to remove from the script

**Comment by petemill** [comment]
@mihaiplesa @bsclifton amended to be very simple. It now simply removes the call to exit the script.

Removes:
```
    if (config.buildConfig === 'Release') {
      process.exit(1)
    }
```

File: lib/build.js (hunk 1 above)
Comment (petemill) [reply lib/build.js]: removed

File: lib/build.js (hunk 2 above)
Comment (petemill) [reply lib/build.js]: removed

File: Jenkinsfile (hunk 3 above)
Comment (petemill) [reply Jenkinsfile]: removed

**Pull request merged**